gen/gen.go

```diff
@@ -128,18 +128,47 @@ func Gen(exercise string, j interface{}, t *template.Template) error {
 		Version: commonMetadata.Version,
 	}, j}
 
+	// Use two potential output files, one for success, and one for format.Source failure.
+	casesFileName := filepath.Join(dirExercise, "cases_test.go")
+	casesErrorFileName := filepath.Join(dirExercise, "cases_test.go-error")
```

Comment:
Yes, that makes sense, I'm not set on this by any means. I wonder though if someone does want to access it as a normal go file and so they change the extension to `.go` and then it's not protected from committing by the `.gitignore` file, but then I don't suppose we're going to accidentally let a commit with a failing test file into the codebase? 
Another idea might be to write the error output to `stdout` and then it's open to piping to a file if so desired? This does then put the onus on the committer to not version control it. I've certainly accidentally version controlled files I don't want to before, maybe that's more trouble. Just thinking aloud here really. :smile: